drivers/SmartThings/matter-thermostat/src/init.lua

```diff
@@ -502,8 +700,14 @@ local function do_configure(driver, device)
       end
     end
     profile_name = profile_name .. create_air_quality_sensor_profile(device)
-
-  elseif #thermostat_eps > 0 then
+  elseif device_type == WATER_HEATER_DEVICE_TYPE_ID then
+    -- If a Water Heater is composed of Electrical Sensor device type, it must support both ElectricalEnergyMeasurement and
+    -- ElectricalPowerMeasurement clusters.
+    local electrical_sensor_eps = device:get_endpoints(clusters.ElectricalEnergyMeasurement.ID)
+    if #electrical_sensor_eps > 0 then
+      profile_name = "water-heater-power-energy-powerConsumption"
+    end
+  elseif #thermostat_eps > 0 and device_type ~= HEAT_PUMP_DEVICE_TYPE_ID then
```

Comment:
I think we should keep this simple and implement it as:
```suggestion
  elseif device_type == HEAT_PUMP_DEVICE_TYPE_ID then
    profile_name = "heat-pump"
  elseif #thermostat_eps > 0 then
```
This is more extensible and keeps the original logical flow.